tests/GrapheneOS/ConnectivityCheckSettingTest/src/grapheneos/test/ConnectivityCheckSettingTest.java

```diff
@@ -0,0 +1,147 @@
+package grapheneos.test;
+
+import com.android.tradefed.util.RunUtil;
+
+import com.android.tradefed.device.INativeDevice;
+import com.android.tradefed.testtype.DeviceJUnit4ClassRunner;
+import com.android.tradefed.testtype.junit4.BaseHostJUnit4Test;
+import com.android.tradefed.device.DeviceNotAvailableException;
+
+import com.android.tradefed.device.CollectingOutputReceiver;
+import com.android.tradefed.result.InputStreamSource;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+
+@RunWith(DeviceJUnit4ClassRunner.class)
+public class ConnectivityCheckSettingTest extends BaseHostJUnit4Test {
+    private static final int GRAPHENEOS_SERVER_SETTING = 0;
+    private static final int STANDARD_SERVER_SETTING = 1;
+    private static final int DISABLED_SETTING = 2;
+
+    private static final String SYS_PROP = "persist.sys.connectivity_checks";
+    private static final String WIFI_CMD = "cmd wifi set-wifi-enabled %s";
+    private static final String LOG_TAG = "NetworkMonitor";
+    private static final String FILTER_REQ_REGEXP = "PROBE_HTTP|PROBE_HTTPS";
+    private static final String FILTER_DISABLED_REGEXP = "Validation disabled";
+
+    enum ConnectivityCheckServer {
+        GRAPHENEOS("http://connectivitycheck.grapheneos.network/generate_204", "https://connectivitycheck.grapheneos.network/generate_204"),
+        STANDARD("http://connectivitycheck.gstatic.com/generate_204", "https://www.google.com/generate_204");
+
+        public final String httpUrl;
+        public final String httpsUrl;
+
+        ConnectivityCheckServer(String httpUrl, String httpsUrl) {
+            this.httpUrl = httpUrl;
+            this.httpsUrl = httpsUrl;
+        }
+    }
+
+    private void toggleWifi(String state) throws DeviceNotAvailableException {
+        String cmd = String.format(WIFI_CMD, state);
+        getDevice().executeShellV2Command(cmd);
+    }
+
+    @Before
+    public void setUp() throws DeviceNotAvailableException {
+        if (!getDevice().isAdbRoot()) {
+            try {
+                getDevice().enableAdbRoot();
+            } finally {
+                fail("adb root access is required");
```

Comment:
`finally` block runs unconditionally, which breaks the test if adb is not running as root